src/modules/code-block/code-block.component.ts

```diff
@@ -14,26 +15,74 @@ export class StacheCodeBlockComponent implements AfterViewInit {
   public code: string;
 
   @Input()
-  public languageType = 'markup';
+  public set languageType(value: string) {
+    if (this.validLanguages.indexOf(value) > -1) {
+      this._languageType = value;
+    } else {
+      this._languageType = this.defaultLanguage;
+    }
+  }
+  public get languageType(): string {
+    return this._languageType;
+  };
 
   @ViewChild('codeFromContent')
   public codeTemplateRef: any;
 
   public output: string;
+  private readonly defaultLanguage: string = 'markup';
+  private readonly validLanguages: string[] = [
```

Comment:
We are using prism for this right? Do they have some sort of list of supported languages that we can import or check against? Should we put this in a service? Having the languages hardcoded feels weird.